PYTHON_STYLE_GUIDE.md

```diff
@@ -0,0 +1,79 @@
+
+BigchainDB uses Python 3.4+, so you can ignore all PEP8 guidelines specific to Python 2.
+
+### Python Docstrings
+
+PEP8 says some things about docstrings, but not what to put in them or how to structure them. [PEP257](https://www.python.org/dev/peps/pep-0257/) was one proposal for docstring conventions, but we prefer [Google-style docstrings](https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments) instead: they're easier to read and the [napoleon extension](http://www.sphinx-doc.org/en/stable/ext/napoleon.html) for Sphinx lets us turn them into nice-looking documentation. Here are some references on Google-style docstrings:
+
+* [Google's docs on Google-style docstrings](https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments)
+* [napoleon's docs include an overview of Google-style docstrings](http://sphinxcontrib-napoleon.readthedocs.org/en/latest/index.html)
+* [Example Google-style docstrings](http://sphinxcontrib-napoleon.readthedocs.org/en/latest/example_google.html) (from napoleon's docs)
+
+### Maximum Line Length
+
+PEP8 has some [maximum line length guidelines](https://www.python.org/dev/peps/pep-0008/#id17), starting with "Limit all lines to a maximum of 79 characters" but "for flowing long blocks of text with fewer structural restrictions (docstrings or comments), the line length should be limited to 72 characters."
+
+We discussed this at length, and it seems that the consensus is: try to keep line lengths less than 79/72 characters, unless you have a special situation where longer lines would improve readability. (The basic reason is that 79/72 works for everyone, and BigchainDB is an open source project.)
+
+### Single or Double Quotes?
+
+Python lets you use single or double quotes. PEP8 says you can use either, as long as you're consistent. We try to stick to using single quotes, except in cases where using double quotes is more readable. For example:
+```python
+print('This doesn\'t look so nice.')
+print("Doesn't this look nicer?")
+```
+
+### Breaking Strings Across Multiple Lines
+
+Should we use parentheses or slashes (`\`) to break strings across multiple lines, i.e.
+```python
+my_string = ('This is a very long string, so long that it will not fit into just one line '
+             'so it must be split across multiple lines.')
+# or
+my_string = 'This is a very long string, so long that it will not fit into just one line ' \
+            'so it must be split across multiple lines.'
+```
+
+It seems the preference is for slashes, but using parentheses is okay too. (There are good arguments either way. Arguing about it seems like a waste of time.)
+
+### Using the % operator or `format()` to Format Strings
+
+Given the choice:
+```python
+x = 'name: %s; score: %d' % (name, n)
+# or
+x = 'name: {}; score: {}'.format(name, n)
+```
+
+we use the `format()` version. It's newer and doesn't require you to remember a bunch of arcane markup.
```

Comment:
It may make sense to simply quote Pyrhon's documentation (https://docs.python.org/2/library/stdtypes.html#str.format) here:

> This method of string formatting is the new standard in Python 3, and should be preferred to the `%` formatting described in [String Formatting Operations](https://docs.python.org/2/library/stdtypes.html#string-formatting) in new code.